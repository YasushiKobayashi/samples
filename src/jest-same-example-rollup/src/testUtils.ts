const testUtils = () => {
  return 'test utils method'
}

export default testUtils
