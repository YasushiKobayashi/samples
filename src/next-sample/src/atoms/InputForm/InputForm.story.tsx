import * as React from 'react'
import type { Meta, StoryFn } from '@storybook/react'
import { expect, userEvent, within } from '@storybook/test'
import * as test from '@storybook/test'
import { waitFor } from '@testing-library/react'

import { InputForm } from './InputForm'

export default {
  title: 'atoms/InputForm',
  component: InputForm,
} as Meta<typeof InputForm>

const base = {
  id: 'id',
  label: 'label',
  val: '',
  onChange: test.fn(),
}

const PrimaryTemplate: StoryFn<typeof InputForm> = additionalProps => {
  base.onChange.mockClear()
  const props = { ...base, ...additionalProps }
  return <InputForm {...props} />
}

export const Primary = PrimaryTemplate.bind({})
Primary.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement)
  await waitFor(async () => {
    await userEvent.type(canvas.getByLabelText('label'), 'v')
  })
  expect(base.onChange).toHaveBeenCalledWith('v')
}
