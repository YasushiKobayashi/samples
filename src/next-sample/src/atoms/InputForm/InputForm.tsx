import * as React from 'react'

interface Props {
  id: string
  label: string
  val: string
  onChange: (v: string) => void
}

export const InputForm: React.FC<Props> = ({ id, label, val, onChange }) => {
  return (
    <div>
      <label htmlFor={id}>{label}</label>
      <input type="text" id={id} value={val} onChange={e => onChange(e.currentTarget.value)} />
    </div>
  )
}
