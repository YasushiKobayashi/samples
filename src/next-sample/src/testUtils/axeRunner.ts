import { configureAxe } from 'jest-axe'

export const axeRunner = configureAxe({
  rules: {},
})
