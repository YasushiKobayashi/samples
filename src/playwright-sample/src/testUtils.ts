import AxeBuilder from '@axe-core/playwright'

export const baseUrl = 'http://localhost:3000/'
export const baseDir = 'test-results/screenshots'

const axeRunner = async (page: Page, disableRules: string[] = []) => {
  const results = await new AxeBuilder({ page })
    .disableRules(['image-alt', 'color-contrast', 'meta-viewport', 'link-name', ...disableRules])
    .analyze()

  if (results.violations.length > 0) {
    const title = await page.title()
    const url = page.url()
    console.error(title, url, results.violations)
    // test.fail()
  }
}

let i = 1
export const testPage = async (page: Page, basePath: string, disableRules: string[] = []) => {
  await axeRunner(page, disableRules)
  // eslint-disable-next-line no-plusplus
  await page.screenshot({ path: `${basePath}/${i++}.png`, fullPage: true })
}
