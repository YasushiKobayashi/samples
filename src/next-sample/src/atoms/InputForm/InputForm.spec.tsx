import * as React from 'react'
import { composeStories } from '@storybook/react'
import { act, cleanup, render } from '@testing-library/react'

import { axeRunner } from '@/testUtils/axeRunner'

import * as stories from './InputForm.story'

const { Primary } = composeStories(stories)

describe('atoms/InputForm', () => {
  afterEach(() => {
    cleanup()
  })

  it('Snap Shot', async () => {
    const { container, asFragment } = render(<Primary />)
    expect(asFragment()).toMatchSnapshot()

    await act(async () => {
      const results = await axeRunner(container)
      expect(results).toHaveNoViolations()
    })

    await Primary.play({ canvasElement: container })
  })
})
