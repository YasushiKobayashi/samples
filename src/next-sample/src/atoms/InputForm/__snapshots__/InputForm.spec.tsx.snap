// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/InputForm Snap Shot 1`] = `
<DocumentFragment>
  <div>
    <label
      for="id"
    >
      label
    </label>
    <input
      id="id"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
